src/main/java/com/spotify/reaper/storage/MemoryStorage.java

```diff
@@ -496,4 +497,40 @@ public int hashCode() {
     }
   }
 
+  @Override
+  public boolean takeLeadOnSegment(long segmentId) {
+    return true;
+  }
+
+  @Override
+  public boolean renewLeadOnSegment(long segmentId) {
+    return true;
+  }
+
+  @Override
+  public void releaseLeadOnSegment(long segmentId) {
+  }
+
+  @Override
+  public void storeHostMetrics(HostMetrics hostMetrics) {
+  }
+
+  @Override
+  public Optional<HostMetrics> getHostMetrics(String hostName) {
+    return Optional.absent();
+  }
+
+  @Override
+  public StorageType getStorageType() {
+    return StorageType.MEMORY;
+  }
+
+  @Override
+  public int countRunningReapers() {
+    return 1;
+  }
+
+  @Override
+  public void saveHeartbeat() {
```

Comment:
No we just want to skip those parts on backends that do not support fault tolerance.
I'll add comments to explain that, since I don't want it to fail.